src/BaseHostPlugin.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Allow to modify the the schema and the host of an existing request.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class BaseHostPlugin implements Plugin
+{
+    /**
+     * @var UriInterface
+     */
+    private $baseUri;
+
+    /**
+     * @var bool
+     */
+    private $replace;
+
+    /**
+     * @param UriInterface $baseUri
+     * @param array        $config {
+     *     @var bool $replace True will replace all hosts, false will only add host when none is specified.
+     * }
+     */
+    public function __construct(UriInterface $baseUri, array $config = [])
+    {
+        $this->baseUri = $baseUri;
+
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
```

Comment:
Yeah, I saw your changes on the other plugins. I was suggested to do like this on the CachePlugin and I feel it is easer to read this way. But as you say, it is a matter of taste. We use both ways at the moment.